Cannot use 'if' directly on boolean property

Consider the following template fragment `{% if devMode %}`, where `devMode` is a boolean variable in the context.

When compiling the template, the following error is thrown:

```
11:10:33.552 [qtp794994886-36] ERROR c.m.pebble.compiler.CompilerImpl - Error on line 41 in /com/mitchellbosecke/pebble/template/compiled/PebbleTemplatecd00f29e3267073f727b10ed71070410.java:41: error: incompatible types
            if (context.get("devMode")) {
                           ^
  required: boolean
  found:    java.lang.Object
11:10:33.552 [qtp794994886-36] ERROR c.m.pebble.compiler.CompilerImpl - Error on line -1 in Note: /com/mitchellbosecke/pebble/template/compiled/PebbleTemplatecd00f29e3267073f727b10ed71070410.java uses unchecked or unsafe operations.
11:10:33.553 [qtp794994886-36] ERROR c.m.pebble.compiler.CompilerImpl - Error on line -1 in Note: Recompile with -Xlint:unchecked for details.
com.mitchellbosecke.pebble.error.CompilationException: Compilation error occurred(null:null)
    at com.mitchellbosecke.pebble.compiler.CompilerImpl.instantiateTemplate(CompilerImpl.java:207)
    at com.mitchellbosecke.pebble.PebbleEngine$1.call(PebbleEngine.java:184)
    at com.mitchellbosecke.pebble.PebbleEngine$1.call(PebbleEngine.java:151)
    at java.util.concurrent.FutureTask.run(FutureTask.java:262)
    at com.mitchellbosecke.pebble.cache.DefaultTemplateLoadingCache.get(DefaultTemplateLoadingCache.java:28)
    at com.mitchellbosecke.pebble.PebbleEngine.compile(PebbleEngine.java:151)
```

Should this usage of 'if' be disallowed? If so, a better error should be thrown.

--- a/src/main/java/com/mitchellbosecke/pebble/node/NodeIf.java
+++ b/src/main/java/com/mitchellbosecke/pebble/node/NodeIf.java
@@ -40,7 +40,8 @@ public class NodeIf extends AbstractNode {
 				compiler.newline().write("if (");
 			}
 
-			compiler.subcompile(ifStatement.getLeft()).raw(") {").newline().indent().subcompile(ifStatement.getRight());
+			compiler.raw("(Boolean) ").subcompile(ifStatement.getLeft()).raw(") {").
+			        newline().indent().subcompile(ifStatement.getRight());
 
 		}
 

--- a/src/test/java/com/mitchellbosecke/pebble/CoreTagsTest.java
+++ b/src/test/java/com/mitchellbosecke/pebble/CoreTagsTest.java
@@ -50,6 +50,22 @@ public class CoreTagsTest extends AbstractTest {
 		assertEquals("no", writer.toString());
 	}
 
+    @Test
+    public void testIfWithDirectProperty() throws PebbleException, IOException {
+        Loader loader = new StringLoader();
+        PebbleEngine pebble = new PebbleEngine(loader);
+        pebble.setStrictVariables(false);
+
+        String source = "{% if variable %}yes{% else %}no{% endif %}";
+        PebbleTemplate template = pebble.compile(source);
+        Map<String, Object> context = new HashMap<>();
+        context.put("variable", true);
+
+        Writer writer = new StringWriter();
+        template.evaluate(writer, context);
+        assertEquals("yes", writer.toString());
+    }
+
 	@Test
 	public void testFlush() throws PebbleException, IOException {
 		Loader loader = new StringLoader();

